src/rules/unboundFunctionThisRule.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../lint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "disallowed \"this\" outside of class body" ;
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new UnboundFunctionThisWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class UnboundFunctionThisWalker extends Lint.ScopeAwareRuleWalker<{inClass: boolean}> {
+    public createScope(node: ts.Node): { inClass: boolean } {
+        let isClassScope = node.kind === ts.SyntaxKind.ClassDeclaration || node.kind === ts.SyntaxKind.ClassExpression;
+
+        return {
+            inClass: isClassScope,
+        };
+    }
+
+    protected validateThisKeyword(node: ts.Node) {
+        if (!this.getAllScopes().some(scope => scope.inClass)) {
```

Comment:
Normally as a style guideline we surround arrow func arguments with parens:
`(scope) => scope.inClass`. Just makes them easier to distinguish visually. This should be a lint rule!